src/reducers/schemaReducer.js

```diff
@@ -10,9 +16,28 @@ const INITIAL_STATE = Map({
 export default (state = INITIAL_STATE, { type, payload }) => {
   switch (type) {
     case FLAG_GETTING_SCHEMAS:
-      return state.update('activity', updateActivityList(payload));
+    case FLAG_SETTING_SCHEMA:
+    case FLAG_DELETING_SCHEMA: {
+      if (payload) {
+        return state.update('activity', (list) => list.push(type));
```

Comment:
Another solution I could suggest is to have 
`activity: { [FLAG_DELETING_SCHEMA]: [], ... }` and use the same `updateActivityList` function to update it. Then if you need this activity you can flatten the `activity` map and check for a `true` value.

Reply:
I love it! I feel like every reducer should use that logic.